src/charm.py

```diff
@@ -197,17 +200,15 @@ def _reconcile(self) -> None:
             # by the _validate_state method
             return
 
-        if self.certificates._tls_relation_created():  # pylint: disable=protected-access
-            # Reconcile certificates in case the certificates relation is present
-            tls_information = TLSInformation.from_charm(self, self.certificates)
-            self._tls.certificate_available(tls_information)
-
         ha_information = HAInformation.from_charm(self)
         self._reconcile_ha(ha_information)
 
         config = CharmConfig.from_charm(self)
         match proxy_mode:
             case ProxyMode.INGRESS:
+                tls_information = TLSInformation.from_charm(self, self.certificates)
```

Comment:
As far as I can see you are executing these two lines on every case
```
 tls_information = TLSInformation.from_charm(self, self.certificates)
self._tls.certificate_available(tls_information)
```
Can you move them out of the match-case?